runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPlugin.java

```diff
@@ -0,0 +1,1365 @@
+		deathSpot = actor.getWorldLocation();   						// Death Location
+		playerLocation = client.getLocalPlayer().getWorldLocation(); 	// Player Location on NPC Death
+		deathName = actor.getName();									// NPC Name
+		deathID = npc.getId();											// NPC ID
+		lastBossKilled = actor.getName();								// NPC Name (stored for pets)
+
+		// Get possible death locations
+		deathLocations = getExpectedLootPoints(npc, deathSpot);
+		// Loop over each death location and store its current item contents
+		for (WorldPoint point : deathLocations)
+		{
+			if (point != null)
+			{
+				Tile tile = getLootTile(point);
+				Map<Integer, Integer> items = createItemMap(tile.getItemLayer());
+				if (items.size() > 0)
+					itemArray.put(point, items);
+			}
+		}
+
+		// Start watching for NPC Despawn
+		watching = true;
+		if (deathName.equals("Zulrah"))
+			watchingItemLayers = true;
+	}
+
+	@Subscribe
+	public void onActorDespawn(ActorDespawned despawned)
+	{
+		Actor actor = actors.get(despawned.getActor());
+		if (actor != null)
+		{
+			actors.remove(actor);
+		}
+
+		if (watching)
+		{
+			Actor npc = despawned.getActor();
+			if (npc.getName().equals(deathName) && npc.getWorldLocation().equals(deathSpot))
+			{
+				// Correct Boss Despawned
+				watching = false;
+				// Find the drops from the correct tile and return them in the correct format
+				ArrayList<DropEntry> drops = createDropEntryArray((NPC) npc);
+				// Specific use case(s)
+				String npcName = npc.getName();
+				if (npcName.equals("Dusk"))
+					npcName = "Grotesque Guardians";
+
+				if (drops != null)
+				{
+					// Add LootEntry by Boss Name
+					AddBossLootEntry(npcName, drops);
+				}
+				else
+				{
+					log.debug("Error creating DropEntry array");
+				}
+
+				// Reset Variables
+				deathSize = -1;			// NPC Size
+				deathSpot = null;		// NPC Death WP
+				playerLocation = null;	// Player WP on NPC Death
+				deathName = null;		// NPC Name
+				deathID = -1;			// NPC ID
+
+				// Clear maps used for creating the DropEntry array
+				changedItemLayerTiles.clear();	// Changed Tiles
+				deathLocations = null;			// Possible Loot Tiles
+				itemArray.clear();				// Items on Possible Loot Tiles
+			}
+		}
+	}
+
+	@Subscribe
+	public void onItemLayerChanged(ItemLayerChanged event)
+	{
+		if (watchingItemLayers)
+		{
+			changedItemLayerTiles.add(event.getTile());
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		// Only update if our plugin config was changed
+		if (!event.getGroup().equals("bosslogger"))
+		{
+			return;
+		}
+
+		handleConfigChanged(event.getKey());
+	}
+
+	// Searches chat messages for kill count value or pet drop
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SERVER && event.getType() != ChatMessageType.FILTERED)
+		{
+			return;
+		}
+
+		String chatMessage = event.getMessage();
+
+		// Barrows KC
+		if (chatMessage.startsWith("Your Barrows chest count is"))
```

Comment:
Rather than doing a string comparison here, you could instead use a regular expression as you do below that will capture the kill value - If you get a returned value then you know the chat message started in the correct form.

Minor, but means you just apply a regex, rather than string comparison and a regex.